src/pages/offer-details/OfferDetails.jsx

```diff
@@ -1,7 +1,49 @@
+import { useState, useEffect } from 'react'
 import PageWrapper from '~/components/page-wrapper/PageWrapper'
+import HeaderCard from '~/components/header-card/HeaderCard'
+import AboutOffer from '~/components/about-offer/AboutOffer'
+import offerIcon from '~/assets/img/offer-page/offer-icon.svg'
+import { useSelector } from 'react-redux'
+import { useLocation } from 'react-router-dom'
+import { offerService } from '~/services/offer-service'
 
 const OfferDetails = () => {
-  return <PageWrapper>OfferDetails Page</PageWrapper>
+  const { pathname } = useLocation()
+  const codedOfferId = pathname.split('/')[2]
+
+  const [offer, setOffer] = useState(null)
```

Comment:
It will be better to use `{}` as a default state to maintain the same data type that we expect from the received data.